src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -635,25 +635,16 @@ private BoundNode IntroduceFrame(BoundNode node, LambdaFrame frame, Func<ArrayBu
 
             var prologue = ArrayBuilder<BoundExpression>.GetInstance();
 
-            BoundExpression newFrame;
-            if (frame.Constructor == null)
-            {
-                Debug.Assert(frame.TypeKind == TypeKind.Struct);
-                newFrame = new BoundDefaultExpression(syntax: syntax, type: frameType);
```

Comment:
no difference when the frame is a struct. Frame is an instance and pointer is a variable for it.  

In a class case we have to create a new frame instance every time the scope is entered and repoint FP to the new instance. That is because lifetime of the closure could exceed lifetime of the scope and more than one frame instance could be alive at the same time.

We do not need to create a _new instance_ of a struct frame though since the frame lives on the stack. It was created when we entered the containing method.

The only thin this code is doing is zeroing-out that struct, which is redundant. Any variable backed by that struct will be assigned something before it is used.

Frame pointer is a bit of confusing name, now that frames can be structs.